native/cocos/platform/filesystem/FilePath.h

```diff
@@ -0,0 +1,67 @@
+
+/****************************************************************************
+ Copyright (c) 2022 Xiamen Yaji Software Co., Ltd.
+ 
+ http://www.cocos.com
+ 
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated engine source code (the "Software"), a limited,
+ worldwide, royalty-free, non-assignable, revocable and non-exclusive license
+ to use Cocos Creator solely to develop games on your target platforms. You shall
+ not use Cocos Creator software for developing other software or tools that's
+ used for developing games. You are not granted to publish, distribute,
+ sublicense, and/or sell copies of Cocos Creator.
+ 
+ The software or tools in this License Agreement are licensed, not sold.
+ Xiamen Yaji Software Co., Ltd. reserves all rights not expressly granted to you.
+ 
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+ ****************************************************************************/
+
+#pragma once
+
+#include "base/Macros.h"
+#include <iostream>
+#include "base/std/container/string.h"
+#include "base/std/container/vector.h"
+
+namespace cc {
+
+class CC_DLL FilePath {
+public:
+    FilePath();
+    FilePath(const FilePath& that);
+    FilePath(const ccstd::string& path);
+    ~FilePath();
+    
+    FilePath& operator=(const FilePath& that);
+    bool operator==(const FilePath& that) const;
+    const char& operator[](int i) const;
+
+    const ccstd::string& value() const { return _path; }
+    bool empty() const { return _path.empty(); };
+    void clear() { _path.clear(); }
+
+    FilePath baseName() const;
+    FilePath dirName() const;
+    void removeLastSeparator();
+
+    ccstd::string finalExtension(bool tolower = true) const;
```

Comment:
What's `final extension`?

Reply:
The extension may contain multiple, linux-like compressed files
Example: 'package.tar.gz', 'final extension' refers to 'gz', not 'tar.gz'

Reply:
Is it a common sense?

Reply:
Most files have only one extension. May need to be obtained separately in the future